autopush-common/src/reliability.rs

```diff
@@ -179,33 +169,59 @@ impl PushReliability {
         conn: &mut C,
         expr: u64,
     ) -> Result<()> {
-        let purged: Vec<String> = conn.zrangebyscore(EXPIRY, 0, expr as isize)?;
+        // First, get the list of values that are to be purged.
+        let mut purge_pipe = redis::Pipeline::new();
+        purge_pipe.zrangebyscore(EXPIRY, 0, expr as isize);
+        let result_list: Vec<Vec<redis::Value>> = purge_pipe.query_async(conn).await?;
```

Comment:
it looks like you can keep the previous, simpler single call by specifying this method takes a `C: ConnectionLike + AsyncCommands`
```suggestion
        let purged: Vec<String> = conn.zrangebyscore(EXPIRY, 0, expr as isize).await?;
```